pkg/executor/infoschema_reader.go

```diff
@@ -898,6 +897,7 @@ func (e *hugeMemTableRetriever) dataForColumnsInTable(ctx context.Context, sctx
 			internalCtx := kv.WithInternalSourceType(context.Background(), kv.InternalTxnOthers)
 			// Build plan is not thread safe, there will be concurrency on sessionctx.
 			if err := runWithSystemSession(internalCtx, sctx, func(s sessionctx.Context) error {
+				is := sessiontxn.GetTxnManager(s).GetTxnInfoSchema()
```

Comment:
is this where the fix is, replace with another session ctx?

Reply:
The original error is "ctx provider not found", but it's overwritten https://github.com/pingcap/tidb/pull/54809. So the first fix is add `sessiontxn.NewTxn`.

Then another problem is the system session still uses the `is` captured by closure, which has conflict with user session. So moving it here.